drivers/can/can_nxp_s32_canxl.c

```diff
@@ -152,7 +152,7 @@ static int can_nxp_s32_get_capabilities(const struct device *dev, can_mode_t *ca
 
 	*cap = CAN_MODE_NORMAL | CAN_MODE_LOOPBACK | CAN_MODE_LISTENONLY;
 
-#if CONFIG_CAN_FD_MODE
+#if defined(CONFIG_CAN_FD_MODE) && defined(CONFIG_CAN_NXP_S32_RX_FIFO)
```

Comment:
This will leave a lot of unused code paths compiled into this driver when either `CONFIG_CAN_FD_MODE` or `CONFIG_CAN_NXP_S32_RX_FIFO` is enabled, but the other one isn't.

Wouldn't it make more sense to determine whether this driver supports CAN FD at all (depends on `CONFIG_CAN_FD_MODE=y` && `CONFIG_CAN_NXP_S32_RX_FIFO=y`) at the very top of the file? Create a new define for this and replace all in-driver use of `#ifdef CONFIG_CAN_FD_MODE` with this?

That way, if either `CONFIG_CAN_FD_MODE=n` or `CONFIG_CAN_NXP_S32_RX_FIFO=n`, the driver will be compiled to only support classic CAN and no unused code paths will be included.

Reply:
Thanks for nice suggestion. I updated this.